exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -177,6 +182,18 @@ public void run() {
     });
   }
 
+  /** If we are in distributed mode, disallow using loopback address to register new drillbit
+   *  unless zookeeper is on the same node with drillbit. */
+  private void checkLoopbackAddress(String address) throws DrillbitStartupException, UnknownHostException {
+    if (isDistributedMode) {
+      String coordAddress = StringUtils.substringBefore(config.getString(ExecConstants.ZK_CONNECTION), ":");
+      if (!InetAddress.getByName(coordAddress).isLoopbackAddress()
```

Comment:
At some point, we wanted to allow loopback only if we have one node only but it's much safer to disallow loopback in distributed mode at all. Updated pull request.